Zend/zend_compile.c

```diff
@@ -9150,11 +9146,6 @@ void zend_eval_const_expr(zend_ast **ast_ptr) /* {{{ */
 				zend_error_noreturn(E_COMPILE_ERROR, "Cannot use [] for reading");
 			}
 
-			if (ast->attr & ZEND_DIM_ALTERNATIVE_SYNTAX) {
-				ast->attr &= ~ZEND_DIM_ALTERNATIVE_SYNTAX; /* remove flag to avoid duplicate warning */
-				zend_error(E_DEPRECATED, "Array and string offset access syntax with curly braces is deprecated");
-			}
```

Comment:
I'd prefer to retain this code and generate an E_COMPILE_ERROR here, so we can provide a better error message. We can drop the actual grammar if/when we run into conflicts.

Reply:
Done.

Reply:
Can you please add tests for the two errors (one in constexpr context, one outside)?

Reply:
ACK.